src/integrationTest/groovy/CompilationAndGeneration.groovy

```diff
@@ -0,0 +1,110 @@
+import org.junit.*
+import static org.junit.Assert.assertTrue
+import static org.junit.Assert.fail
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.api.Project
+import org.gradle.tooling.GradleConnector
+import org.gradle.tooling.ProjectConnection
+import org.gradle.tooling.BuildLauncher
+import java.util.Map
+import java.util.HashMap
+import cruise.umple.util.SampleFileWriter;
+
+public class CompilationAndGeneration {    
+    private static final DIRECTORIES_ROOT = "src/integrationTest/resources/TestProject/"
+    private static final Map<String, ArrayList<String>> DIRECTORIES_TO_CHECK
+    // Key = path to directory we want to check, value = the files we expect to find in the directory
+    static {
+        Map<String, ArrayList<String>> aMap = new HashMap<String, ArrayList<String>>()
```

Comment:
I haven't been able to figure out how to make an immutable `ArrayListMultimap`. What I've got right now probably isn't what you're looking for:

```
private static final ImmutableMultimap<String, ArrayList<String>> DIRECTORIES_TO_CHECK
    // Key = path to directory we want to check, value = the files we expect to find in the directory   
    static {
        ImmutableMultimap.Builder<String, ArrayList<String>> builder = ImmutableMultimap.builder();
	    builder.put(DIRECTORIES_ROOT + "subproj/build/classes/generatedSource", ["Subproj.class", "Subproj2.class"])
        builder.put(DIRECTORIES_ROOT + "sub2/build/classes/generatedSource", ["Sub2.class", "Sub22.class"])
        builder.put(DIRECTORIES_ROOT + "subproj/generated/java", ["Subproj.java", "Subproj2.java"])
        builder.put(DIRECTORIES_ROOT + "sub2/generated/java", ["Sub2.java", "Sub22.java"])
	    DIRECTORIES_TO_CHECK = builder.build();
    }
```

I don't think this is all that much of an improvement. Since the map is immutable the usecase for a` multimap` and a `hashmap` don't seem that different.